### PR TITLE
test: add fuzz tests for column data type alteration

### LINKED ISSUE #3973: Add fuzz tests for column data type alteration
### What type of enhancement is this?

Tech debt reduction

### What does the enhancement do?

We support to alter column data type in #3517, Let's add fuzz tests for this feature. 

See also: 
- #3972
- #3174 
- #3517

### Implementation challenges

_No response_

### PR DESCRIPTION
**Pull request by realtaobo**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## Refer to a related PR or issue link (optional)
#3973 

## What's changed and what's your intention?

Just add fuzz test for `Alter Table <table_name> MODIFY COLUMN <column_name> <column_type>`

## Checklist

- [ ] I have written the necessary rustdoc comments.
- [x] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.

### CODE REVIEW

**Comment by WenyXu** [comment]
Would you like to add the target to CI cfg
https://github.com/GreptimeTeam/greptimedb/blob/f024054ed34f722533110fa43468e1079edf2750/.github/workflows/develop.yml#L144
https://github.com/GreptimeTeam/greptimedb/blob/f024054ed34f722533110fa43468e1079edf2750/.github/workflows/develop.yml#L286

**Comment by realtaobo** [comment]
@WenyXu Sure.
BTW, it seems not work to read env var `GT_MYSQL_ADDR` form .env config file.
https://github.com/GreptimeTeam/greptimedb/blob/f024054ed34f722533110fa43468e1079edf2750/tests-fuzz/src/utils.rs#L38-L41
If I don't set Environment variables，I got `"GT_MYSQL_ADDR is empty, ignores test"`, even though I have a .env file int the root.  How should I solve this problem?
```sh
ProductName:            macOS
ProductVersion:         14.5
BuildVersion:           23F79
```
**Exec cmd:** `cargo fuzz run fuzz_alter_table --fuzz-dir tests-fuzz -D -s none -- -runs=1`

**Comment by WenyXu** [comment]
> @WenyXu It seems unnecessary. I added it to `fuzz_alter_table` as a part of alter table. What do you think？

Cool, I got you

> BTW, it seems not work to read env var GT_MYSQL_ADDR form .env config file.

Interesting, it works on my machine.

> If I don't set Environment variables，I got "GT_MYSQL_ADDR is empty, ignores test", even though I have a .env file int the root. How should I solve this problem?

What does the directory structure and .env file look like? (Don't submit any secrets)

**Comment by realtaobo** [comment]
> What does the directory structure and .env file look like? (Don't submit any secrets)

The content is the same as file [.env.example](https://github.com/GreptimeTeam/greptimedb/blob/main/.env.example), only `GT_MYSQL_ADDR` is changed to:
```sh
GT_MYSQL_ADDR=127.0.0.1:4002
```

**Comment by WenyXu** [comment]
> The content is the same as file [.env.example](https://github.com/GreptimeTeam/greptimedb/blob/main/.env.example), only `GT_MYSQL_ADDR` is changed to:

I don't have any idea about this👀

**Comment by realtaobo** [comment]
> I don't have any idea about this👀

Got it, a little problem, maybe I should restart my machine 🐶

File: tests-fuzz/src/ir.rs [hunk 1]
```diff
@@ -355,6 +355,20 @@ pub fn droppable_columns(columns: &[Column]) -> Vec<&Column> {
...
+/// Returns columns that can use the alter table modify command
+pub fn modifiable_columns(columns: &[Column]) -> Vec<&Column> {
+    columns
+        .iter()
+        .filter(|column| {
+            !column.options.iter().any(|option| {
+                option == &ColumnOption::PrimaryKey
+                    || option == &ColumnOption::TimeIndex
+                    || option == &ColumnOption::NotNull
+            })
```
Comment (WenyXu) [inline tests-fuzz/src/ir.rs:367]: The `Nullable` and `TimeIndex` columns should also be modifiable. cc @evenyag

File: tests-fuzz/src/context.rs [hunk 2]
```diff
@@ -129,6 +129,15 @@ impl TableContext {
                 self.name = new_table_name;
                 Ok(self)
             }
+            AlterTableOperation::ModifyDataType { column } => {
+                for iter in self.columns.iter_mut() {
+                    if iter.name != column.name {
+                        continue;
+                    }
+                    iter.column_type = column.column_type.clone();
+                }
```
Comment (WenyXu) [inline tests-fuzz/src/context.rs:138]: How about using `position`

**Comment by WenyXu** [comment]
We can't to close the #3973 now. We need to insert some values before the alteration and validate the inserted values after the alteration. However, it could be done in another PR.

File: tests-fuzz/src/ir.rs (hunk 1 above)
Comment (realtaobo) [reply tests-fuzz/src/ir.rs:367]: It should not be supported at present.
https://github.com/GreptimeTeam/greptimedb/blob/4e5dd1ebb04d910bdb88611a8f824529701606b8/src/table/src/metadata.rs#L487-L494

Comment (WenyXu) [reply tests-fuzz/src/ir.rs:367]: I see

**Review by WenyXu — APPROVED** [review]

**Review by evenyag — APPROVED** [review]

**Pull request merged**